pacman/model/graphs/machine/machine_spinnaker_link_vertex.py

```diff
@@ -69,7 +69,12 @@ def virtual_chip_y(self):
 
     @overrides(AbstractVirtual.set_virtual_chip_coordinates)
     def set_virtual_chip_coordinates(self, virtual_chip_x, virtual_chip_y):
-        self._virtual_chip_x = virtual_chip_x
-        self._virtual_chip_y = virtual_chip_y
-        self.add_constraint(ChipAndCoreConstraint(
-            self._virtual_chip_x, self._virtual_chip_y))
+        if virtual_chip_x is not None and virtual_chip_y is not None:
+            self._virtual_chip_x = virtual_chip_x
+            self._virtual_chip_y = virtual_chip_y
+            self.add_constraint(ChipAndCoreConstraint(
+                self._virtual_chip_x, self._virtual_chip_y))
+            if (self._app_vertex is not None and
```

Comment:
This assumes that a virtual application vertex only has one X,Y location. That seems wrong-headed.

Reply:
given every external device has been 1 machine vertex to date. its acutally perfectly valid. plus the virutal chips are menat to able to ahndle infinite cores. so def allowed to have 1 chip